runtime/runtime/src/lib.rs

```diff
@@ -1303,12 +1323,19 @@ impl Runtime {
             if let Some(outcome_with_id) = result? {
                 *total_gas_burnt =
                     safe_add_gas(*total_gas_burnt, outcome_with_id.outcome.gas_burnt)?;
+                *total_compute_usage =
+                    safe_add_compute(*total_compute_usage, outcome_with_id.outcome.compute_usage)?;
+                // TODO(#8032): Remove when compute costs are stabilized.
+                debug_assert_eq!(
+                    total_compute_usage, total_gas_burnt,
+                    "Compute usage must match burnt gas"
+                );
                 outcomes.push(outcome_with_id);
             }
             Ok(())
         };
 
-        let gas_limit = apply_state.gas_limit.unwrap_or(Gas::max_value());
+        let compute_limit = apply_state.gas_limit.unwrap_or(Gas::max_value());
```

Comment:
`compute_limit = apply_state.gas_limit` will look confusing for anyone not familiar with the history.

Should we rename `apply_state.gas_limit` to `apply_state.compute_limit` and also change it's type alias to `Option<Compute>`? Or if that just propagates the problem upward, maybe add at least a comment and fix the naming in a later refactoring?

Reply:
I've started doing this in https://github.com/near/nearcore/commit/43a80328514a59cc1f70c728037e4cb603845adc and my conclusion is that fully renaming gas_limit to compute_limit will touch around 230 places (spread across 10s of files).

I anyway think it makes sense to do this renaming, though as above I would try to be a bit more forward-looking here. Concretely, maybe instead of just one limit, we should store `ResourceLimits` struct in `ApplyState` and ultimately in the chunk and make it extensible to support other resource types (Gas, Compute, DIsk IOPS, ...) that we want to limit during the execution of receipts.
For now, I'm adding a comment here and opening a GitHub issue with this refactoring proposal so that we can discuss what is the end-state that we want to achieve here.

Reply:
I've filed https://github.com/near/nearcore/issues/8859 to track this.